### PR TITLE
[MISC] Bump coverage

### PR DESCRIPTION
**Pull request by dragomirp**
* Increase the coverage so that codecov stops complaining
* Update libs (except `kubernetes_service_patch`)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -86,6 +86,9 @@ branch = true
 
 [tool.coverage.report]
 show_missing = true
+exclude_lines = [
+    "logger\\.debug"
+]
```
Comment (dragomirp) [inline pyproject.toml:91]: Excluded on VM: https://github.com/canonical/postgresql-operator/blob/main/pyproject.toml#L84-L86

File: src/patroni.py [hunk 2]
```diff
@@ -476,16 +477,16 @@ def last_postgresql_logs(self) -> str:
...
-        log_files.sort(key=lambda f: f.path, reverse=True)
         try:
+            log_files = container.list_files(POSTGRESQL_LOGS_PATH, pattern=POSTGRESQL_LOGS_PATTERN)
+            if len(log_files) == 0:
+                return ""
+            log_files.sort(key=lambda f: f.path, reverse=True)
             with container.pull(log_files[0].path) as last_log_file:
                 return last_log_file.read()
-        except OSError as e:
+        except Error:
```
Comment (dragomirp) [inline src/patroni.py:487]: I don't think accessing the container will raise an `OSError` and list files will error if the directory doesn't exist.

File: src/relations/postgresql_provider.py [hunk 3]
```diff
@@ -78,9 +78,6 @@ def _on_database_requested(self, event: DatabaseRequestedEvent) -> None:
             event.defer()
             return
 
-        if not self.charm.unit.is_leader():
-            return
-
```
Comment (dragomirp) [inline src/relations/postgresql_provider.py:83]: Dead code, database_requested should only be emitted on the leader: https://github.com/canonical/data-platform-libs/blob/main/lib/charms/data_platform_libs/v0/data_interfaces.py#L2878-L2879

File: tests/unit/test_patroni.py
Comment (dragomirp) [inline tests/unit/test_patroni.py:1]: Mostly copypaste from the VM charm.

Comment (dragomirp) [inline tests/unit/test_patroni.py:1]: Mostly copypaste from the VM charm.

**Review by taurus-forever — APPROVED** [review]
Tnx!

**Review by marceloneppel — APPROVED** [review]
Thanks!

**Pull request merged**